src/web3/saga.test.ts

```diff
@@ -51,31 +51,33 @@ describe(getOrCreateAccount, () => {
       .run()
   })
 
-  it('creates a new account', async () => {
-    const MNEMONIC =
-      'avellana novio zona pinza ducha íntimo amante diluir toldo peón ocio encía gen balcón carro lingote millón amasar mármol bondad toser soledad croqueta agosto'
-    const EXPECTED_ADDRESS = '0xE025583d25Eff2C254999b5904C97bAe9B3F8D83'
-    const EXPECTED_DEK = '0xb6812219f7003c27cc1ef17c2033c033a38cfc52d83f176a0667086787d59d39'
-
-    await expectSaga(getOrCreateAccount)
-      .withState(state)
-      .provide([
-        [select(currentAccountSelector), null],
-        [matchers.call.fn(generateMnemonic), MNEMONIC],
-        [
-          call(storeMnemonic, MNEMONIC, EXPECTED_ADDRESS),
-          {
-            service: 'mnemonic',
-            storage: 'storage',
-          },
-        ],
-        [call(getPasswordSaga, EXPECTED_ADDRESS, false, true), 'somePassword'],
-      ])
-      .put(setAccount(EXPECTED_ADDRESS))
-      .put(setDataEncryptionKey(EXPECTED_DEK))
-      .returns(EXPECTED_ADDRESS)
-      .run()
-  })
+  it.each`
+    expectedAddress                                 | expectedDEK                                                               | mnemonic
+    ${'0xE025583d25Eff2C254999b5904C97bAe9B3F8D83'} | ${'0xb6812219f7003c27cc1ef17c2033c033a38cfc52d83f176a0667086787d59d39'}   | ${'avellana novio zona pinza ducha íntimo amante diluir toldo peón ocio encía gen balcón carro lingote millón amasar mármol bondad toser soledad croqueta agosto'}
+    ${'0x49a4728b4f94e5ccc4ddfe2fe2269ccd431dc052'} | ${'0x0275da8df87a83403d0a0e9aec60e26ac11ebd9764ebe3bbfc5fecb3631e3e0a99'} | ${'Sufrir maniquí índice viral boina reino choza denso gafas aviso nadar glaciar'}
```

Comment:
i spent some time chasing why this test doesn't work, but couldn't exactly figure it out. the tests fail because the seed phrase generates a different wallet address than the one that is expected. however, running on a live environment, the same seed phrase does generate the expected address. from these symptoms you'd assume that we're mocking something wrongly in the tests but i couldn't find which mocks would cause this behaviour.

one thing to note, this test does pass if you use an english seed phrase (e.g. the one we use for the e2e test). 
i'll note what i found and we can decide where to go from there....

the `getOrCreateAccount` saga generates a mnemonic and then gets the private key (+ account address) from the `generateKeysFromMnemonic` function. in the tests we mock generating the mnemonic, but we do not mock any part of generating the keys from the mnemonic. all `generateKeysFromMnemonic` does is decide the derivation path based on number of words in the seed phrase, then call `generateKeys` from the celo lib. the inputs passed in are the mnemonic, bip39, and the derivation path.

i logged these 3 inputs to the `generateKeys` function from the `generateKeysFromMnemonic` function, and compared the results when running the test vs restoring an account in the simulator. the results were the same, but the keys that `generateKeys` returned were different. i then went and put some log statements in the @celo/cryptographic-utils node module, those logs were printed so we are hitting the real function and not mocking it. we are also not mocking bip39 anywhere. `generateKeys` uses bip32 to generate the keys under the hood, we're not mocking that anywhere either. i just cannot understand why the same inputs to this `generateKeys` function is producing different results in a live env vs in the test env. scrolling through our mocks in the mocks folder, i couldn't see any mocks that would obviously interfere with this function.

i've exhausted the timebox i want to spend on this, maybe something in the above sparks some idea for you @MuckT ? we also discussed the value of this test yesterday, if we cover with e2e tests then perhaps it's not worth spending too much time on this unit test?

Reply:
Two things are causing this issue. The first was the the mnemonic in Spanish had accents that were different than what was seen in: https://github.com/bitcoin/bips/blob/master/bip-0039/bip-0039-wordlists.md.
Second there appears to be a mismatch of the DEK's generated by the wallet and those registered on chain. I'll create a follow-up spike to investigate the DEK issue.

Reply:
I had a look the DEK mismatch and it's all good.
We were comparing the public key to the private key, hence the mismatch 🤦 

For `0x49A4728b4F94E5ccC4dDFe2fE2269ccD431dc052` (mnemonic: `avellana novio zona pinza ducha íntimo amante diluir toldo peón ocio encía gen balcón carro lingote millón amasar mármol bondad toser soledad croqueta agosto`, here's the DEK:
- Public key: `0x0334b8d45fc99b433d3f294786d5d32aff0e0ce914f92ff518a42b8854b85e30fb`
- Private key: `0xb6812219f7003c27cc1ef17c2033c033a38cfc52d83f176a0667086787d59d39` 

For `0x49A4728b4F94E5ccC4dDFe2fE2269ccD431dc052` (mnemonic: `sufrir maniquí índice viral boina reino choza denso gafas aviso nadar glaciar`, here's the DEK:
- Public key: `0x0275da8df87a83403d0a0e9aec60e26ac11ebd9764ebe3bbfc5fecb3631e3e0a99`
- Private key: `0x596371d7f1dc3a1dfb9e00b51cb5ca46b73993b962c0fb0e04302004581bf45d`